primitives/core/src/testing.rs

```diff
@@ -64,8 +67,9 @@ impl KeyStore {
 }
 
 #[cfg(feature = "std")]
+#[async_trait]
 impl crate::traits::BareCryptoStore for KeyStore {
```

Comment:
Might be good to simply introduce a blocking wrapper for `BareCryptoStore` instead of sprinkling `block_on` everywhere. You can do that without introducing any new structs, just adding an extra trait that maintains current API something like this:
```rust
pub trait BlockingBareCryptoStore {
  fn keys() -> Vec<...>
}

impl<T: BareCryptoStore> BlockingBareCryptoStore for T {
  fn keys() -> Vec<...> { block_on(BareCryptoStore::keys(self)) }
}
```